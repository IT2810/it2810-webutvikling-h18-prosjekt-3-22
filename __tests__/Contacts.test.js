import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer'
import Contacts from '../src/components/contactHome.js';

/* let findSteps = function (tree, element) {
  let result = undefined;
  let steps = tree.children[2].children[0];
  for (node in steps.children) {
    if (steps.children[node] == element) {
      result = true;
    }
  }
  return result;
}

let findInput = function (tree, element) {
  let result = undefined;
  for (node in tree.children) {
    if (tree.children[node].props.testID == element) {
      result = true;
    }
  }
  return result;
} */

describe('Contact-page test', () => {
  const tree = renderer.create( <
    Contacts / >
  ).toJSON();

  test('Take snapshot of goal-component', async () => {
    await expect(tree).toMatchSnapshot();
  });
/*   test('Verify that stepgoal is 10000', () => {
    let stepGoal = 10000;
    expect(findSteps(tree, stepGoal)).toBe(true);
  });
  test('Verify that steps taken today is 0', () => {
    let stepsToday = 0;
    expect(findSteps(tree, stepsToday)).toBe(true);
  });
  test('Check if inputbox is available', () => {
    expect(findInput(tree, 'goalInput')).toBe(true);
  }); */
});