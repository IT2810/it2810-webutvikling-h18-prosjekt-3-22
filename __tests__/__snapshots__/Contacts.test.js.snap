// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact-page test Take snapshot of goal-component 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "paddingBottom": 15,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#4d79ff",
        "borderBottomColor": "#ddd",
        "borderBottomWidth": 10,
        "justifyContent": "center",
        "marginBottom": 10,
        "width": "100%",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontSize": 18,
          "padding": 26,
        }
      }
    >
       Contacts 
    </Text>
  </View>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={false}
    visible={false}
  >
    <View
      style={
        Object {
          "marginTop": 80,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#4d79ff",
            "borderBottomColor": "#ddd",
            "borderBottomWidth": 10,
            "justifyContent": "center",
            "marginBottom": 10,
            "width": "100%",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "fontSize": 18,
              "padding": 26,
            }
          }
        >
           Add new contact 
        </Text>
      </View>
      <View
        style={
          Object {
            "margin": 20,
          }
        }
      >
        <View
          style={
            Object {
              "borderWidth": 1,
              "flexDirection": "row",
              "marginBottom": 10,
              "padding": 15,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "marginRight": 5,
              }
            }
          >
            Name:
          </Text>
          <TextInput
            allowFontScaling={true}
            onChangeText={[Function]}
            placeholder="Ola Nordmann"
          />
        </View>
        <View
          style={
            Object {
              "borderWidth": 1,
              "flexDirection": "row",
              "marginBottom": 10,
              "padding": 15,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "marginRight": 5,
              }
            }
          >
            Number:
          </Text>
          <TextInput
            allowFontScaling={true}
            onChangeText={[Function]}
            placeholder="123456789"
          />
        </View>
        <View
          style={
            Object {
              "borderWidth": 1,
              "flexDirection": "row",
              "marginBottom": 10,
              "padding": 15,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "marginRight": 5,
              }
            }
          >
            Mail:
          </Text>
          <TextInput
            allowFontScaling={true}
            onChangeText={[Function]}
            placeholder="ola@gmail.com"
          />
        </View>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <View
          accessibilityComponentType="button"
          accessibilityTraits={
            Array [
              "button",
            ]
          }
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "padding": 8,
                    "textAlign": "center",
                  },
                ]
              }
            >
              ✖Close
            </Text>
          </View>
        </View>
        <View
          accessibilityComponentType="button"
          accessibilityTraits={
            Array [
              "button",
            ]
          }
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "padding": 8,
                    "textAlign": "center",
                  },
                ]
              }
            >
              ✔Save
            </Text>
          </View>
        </View>
      </View>
    </View>
  </Modal>
  <View
    accessibilityComponentType="button"
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Add contact
      </Text>
    </View>
  </View>
</View>
`;
